budget-vs-actual/provincial/pipeline-spec.yaml

```diff
@@ -71,113 +73,427 @@ budget-vs-actual-provincial:
             aggregate: sum
         full: true
 
+    ### EPRE 2018-19
     - run: load
       parameters:
+        from: 'https://s3-eu-west-1.amazonaws.com/manual-uploads.vulekamali.gov.za/epre/epre-2018-19-processed-2019-08-23t1245.csv'
         name: epre-2018-19
-        from: 'http://datastore.openspending.org/b9d2af843f3a7ca223eea07fb608e62a/estimates-of-provincial-expenditure-2018-19-uploaded-2019-05-01t1328/final/data/epre-2018-19.csv'
+        format: 'csv'
       cache: true
 
-      # because otherwise string "addition" happens which looks like
-      # '7871520' + '9700802' + '-152' + '100000' = '100000-15297008027871520'
     - run: set_types
       parameters:
         resources: epre-2018-19
         types:
-          value:
+          Value:
             type: number
 
     - run: filter
       parameters:
         resources: epre-2018-19
         in:
-          - finyear: '2018'
+          - FinancialYear: '2018'
+          - BudgetPhase: 'Main Appropriation'
 
     - run: delete_fields
       parameters:
         resources: epre-2018-19
         fields:
-          - voteno
-          - sprogno
-          - subprogramme
+          - VoteNumber
+          - SubprogNumber
+          - Subprogramme
 
     - run: join
       parameters:
         source:
           name: epre-2018-19
           key:
-            - "province"
-            - "department"
-            - "progno"
-            - "programme"
-            - "fg1"
-            - "fg2"
-            - "econ1"
-            - "econ2"
-            - "econ3"
-            - "econ4"
-            - "econ5"
-            - "finyear"
-            - "budget_phase"
+            - "Government"
+            - "Department"
+            - "ProgNumber"
+            - "Programme"
+            - "FunctionGroup1"
+            - "FunctionGroup2"
+            - "EconomicClassification1"
+            - "EconomicClassification2"
+            - "EconomicClassification3"
+            - "EconomicClassification4"
+            - "EconomicClassification5"
+            - "FinancialYear"
+            - "BudgetPhase"
           delete: yes
         target:
           name: epre-2018-19
           key: null
         fields:
-          "province": {}
-          "department": {}
-          "progno": {}
-          "programme": {}
-          "fg1": {}
-          "fg2": {}
-          "econ1": {}
-          "econ2": {}
-          "econ3": {}
-          "econ4": {}
-          "econ5": {}
-          "finyear": {}
-          "budget_phase": {}
-          "value":
+          "Government": {}
```

Comment:
I don't think you need to join for each source - I think you can join at the end, pulling all the different source columns that should end up as one column together in one go https://github.com/vulekamali/treasury-pipelines/blob/master/budget-vs-actual/national/pipeline-spec.yaml#L215

Reply:
Okay, I'm changing the code. I'll test it with one join at the end.

Reply:
Joins deleted.
https://github.com/vulekamali/treasury-pipelines/pull/26/commits/ed4afb10e1d0ffa0442da44e63cf332959d84955